### ISSUE TITLE
Build from existing site-packages

### ISSUE THREAD

**Issue by ipmb** [linkedin/shiv#56]
Would you consider a PR that added a flag like `--from-site-packages` which would copy the contents of an existing `site-packages` directory into the working path?

My use case is that part of my build process requires creating a working virtualenv (needed to generate static files for a Django app). It would be an optimization to just use the virtualenv I already have setup to create the zipapp instead of rebuilding one from scratch.

**Comment by sixninetynine** [comment]
hey @ipmb !

This is a really intriguing idea... How do you foresee it coming to fruition? I'm imagining a flag like `--site-packages` that feeds a conditional here https://github.com/linkedin/shiv/blob/master/src/shiv/cli.py#L119-L123

Something like: 

```
if site_packages and not pip_args:
    ~ copy site packages into working_path ~
else:
    ~ existing logic ~
```

I'm curious if there are any peculiarities with virtualenv's site-packages that would throw a wrench in the mix, but nothing immediately comes to mind. I imagine this could be useful in other scenarios as well (like creating a venv and wanting to "move" that environment to another machine). 

One thing that gives me pause about introducing a feature like this is that it could seriously complicate any efforts towards implementing the functionality mentioned in https://github.com/linkedin/shiv/issues/26, which has been on the back of my mind for a bit.

**Comment by ipmb** [comment]
> I'm imagining a flag like `--site-packages` that feeds a conditional here 

Yep, that's exactly what I was thinking. I'm guessing it's a 5 line change.

> One thing that gives me pause about introducing a feature like this is that it could seriously complicate any efforts towards implementing the functionality mentioned in #26

Maybe the two options could be mutually exclusive? Raising an error if you try to copy site packages and target a different platform?

**Comment by sixninetynine** [comment]
thanks for the PR!

released on pypi @ 0.0.32
